modules/plugin/jdbc/jdbc-oracle/src/main/java/org/geotools/data/oracle/OracleFilterToSQL.java

```diff
@@ -235,11 +243,42 @@ public Object visit(Function function, Object extraData) {
             } catch (IOException ioe) {
                 throw new RuntimeException(IO_ERROR, ioe);
             }
+        } else if (function instanceof JsonPointerFunction) {
+            try {
+                out.write(encodeJsonPointer(function, extraData));
+                return extraData;
+            } catch (IOException ioe) {
+                throw new RuntimeException(IO_ERROR, ioe);
+            }
         }
 
         return super.visit(function, extraData);
     }
 
+    private String encodeJsonPointer(Function jsonPointer, Object extraData) {
+        Expression json = getParameter(jsonPointer, 0, true);
+        Expression pointer = getParameter(jsonPointer, 1, true);
+        if (json instanceof PropertyName && pointer instanceof Literal) {
```

Comment:
Here is also assumes it has to be a straight literal. Ah, another important thing, is when FilterSimplifier is applied in the JDBC code. If it's applied before the splitting, then you'll receive a Literal anyways, as the ``env(...)`` has already been replaced by a Literal in the simplifier.

Reply:
I think the reply to the previous comment also applies here.